open_optical_gating/cli/determine_reference_period.py

```diff
@@ -47,12 +51,15 @@ def establish_indices(sequence, settings):
 
     periods = []
     for i in range(1, len(sequence)):
```

Comment:
Important: surely this loop should be eliminated now we are doing this "on the fly"? We should just be executing this loop body once, for the most recently received frame (sequence[-1])

Reply:
You seem to have removed this so marking ar resolved.